modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/fragment/renderer/HeaderFragmentRenderer.java

```diff
@@ -0,0 +1,234 @@
+import com.liferay.commerce.order.CommerceOrderHttpHelper;
+import com.liferay.commerce.order.content.web.internal.display.context.HeaderFragmentDisplayContext;
+import com.liferay.commerce.order.engine.CommerceOrderEngine;
+import com.liferay.commerce.payment.integration.CommercePaymentIntegrationRegistry;
+import com.liferay.commerce.payment.method.CommercePaymentMethodRegistry;
+import com.liferay.commerce.payment.service.CommercePaymentMethodGroupRelLocalService;
+import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.commerce.service.CommerceOrderService;
+import com.liferay.fragment.renderer.FragmentRenderer;
+import com.liferay.fragment.renderer.FragmentRendererContext;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.configuration.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+import com.liferay.portal.kernel.theme.PortletDisplay;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.util.Locale;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Brian I. Kim
+ */
+@Component(service = FragmentRenderer.class)
+public class HeaderFragmentRenderer implements FragmentRenderer {
+
+	@Override
+	public String getCollectionKey() {
+		return "commerce-product";
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "header");
+	}
+
+	@Override
+	public void render(
+			FragmentRendererContext fragmentRendererContext,
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException {
+
+		/*				Object infoItem = httpServletRequest.getAttribute(
+							InfoDisplayWebKeys.INFO_ITEM);
+
+						if ((infoItem == null) || !(infoItem instanceof CommerceOrder)) {
+							if (_isEditMode(httpServletRequest)) {
+								_printPortletMessageInfo(
+									httpServletRequest, httpServletResponse);
+							}
+
+							return;
+						}*/
+
+		long commerceOrderId = ParamUtil.getLongValues(
+			httpServletRequest,
+			"_com_liferay_commerce_order_content_web_internal_portlet_" +
+				"CommerceOpenOrderContentPortlet_commerceOrderId")[0];
+
```

Comment:
This needs to be investigated for pending orders page vs. placed orders page.

For some reason, when the fragment was placed on pending orders page, the `httpservletRequest` always returned a list of two duplicate `orderId` values which forced me to grab the 0th index value.

However, when the fragment was placed on a placed orders page, the `httpServletRequest` behaved normally and returned a single `orderId`